### ISSUE TITLE
Version numbering of release 2.8

### ISSUE THREAD

**Issue by laalto** [square/picasso#2179]
Updated Picasso version 2.71828 to 2.8. Now Android Lint complains that there is a newer version 2.71828 available. Which is true in semantic versioning sense since 71828 > 8. 

One can suppress the GradleDependency lint check for the library to get rid of the warning. However, I would find it better if version number components were monotonically increasing instead.

**Comment by dimsuz** [comment]
Haha, should've gone for `3.14159`.

**Comment by erikghonyan** [comment]
@jrodbx any thoughts on this or should we go ahead with forcing the version?

**Comment by xanscale** [comment]
just replace versione 2.8 with 2.80000

**Comment by djdance** [comment]
_Failed to resolve: com.squareup.picasso:picasso:2.80000_

what am I doing wrong?

**Comment by xanscale** [comment]
> _Failed to resolve: com.squareup.picasso:picasso:2.80000_
> 
> what am I doing wrong?

no no, is a suggestion for releaser to fix issue

**Comment by djdance** [comment]
I've already got it, but as 2.8, not 2.80000

**Comment by mikezliu** [comment]
Literally unusable

**Comment by kata32** [comment]
From maven, this is funny
![image](https://user-images.githubusercontent.com/26146898/101604978-e79aeb00-3a01-11eb-85da-b8bcec7757ba.png)

**Comment by sczerwinski** [comment]
The same issue for Dependabot on GitHub. I'm getting pull requests like [this](https://github.com/sczerwinski/android-hilt/pull/28) now. Could the future 2.x releases be called e.g. 2.90000 instead of 2.9?

**Comment by shlusiak** [comment]
This is hilarious. Either continue the old scheme or use patch levels or bump to version `3`. I don't think hacks like "2.90000" are going to be free of issues either. That horse has bolted.

**Comment by JakeWharton** [comment]
I'd say integer ordering is pretty deterministic. Not really a hack.

**Comment by kata32** [comment]
> It's been two years and Android Studio still wants me to "upgrade" to 2.71828... can't this be resolved?

Yes, move to [coil](https://github.com/coil-kt/coil), I do that.

**Comment by mikezliu** [comment]
I moved the version to .toml and Android Studio doesn't complain as much (unless you look in project structure)

**Comment by Darksymphony75** [comment]
Same issue, can you please fix the versioning ASAP? Shouldn't be a big issue to fix, thanks

**Comment by kata32** [comment]
> Same issue, can you please fix the versioning ASAP? Shouldn't be a big issue to fix, thanks. I wonder it is still not done for 2 years

It's not that easy to fix, they should use version 3.xxx

**Comment by Darksymphony75** [comment]
> > Same issue, can you please fix the versioning ASAP? Shouldn't be a big issue to fix, thanks. I wonder it is still not done for 2 years
> 
> It's not that easy to fix, they should use version 3.xxx

Yeah, then let's go with that, Can't believe 2 years nothing done with this project

**Comment by zkrige** [comment]
`    implementation("com.squareup.picasso:picasso") { version { strictly "2.8" } }`

**Comment by ryust** [comment]
zkrige... and exactly how does this prevent Android Studio and Gradle from constantly complaining about 2.8?

**Comment by zkrige** [comment]
@ryust when you have `strictly` it knows that you want that version, so it doesn't complain about 2.78128 - there's no warnings

**Comment by ryust** [comment]
Well, Android Studio DOES continue to show there is an issue with it. And what happens when (unlikely) Square updates the release? That is the core problem that should be fixed without users having to monkey around with solutions like this.

**Comment by kurthuwig** [comment]
Yes @zkrige that is correct. But when 2.9 is released, your IDE won't tell you to upgrade, because you've used `strictly`. Using the **e** constant was not the smartest idea and IMHO a version 3.0 should be released to fix this issue.
